client.go

```diff
@@ -70,26 +68,25 @@ type Auth struct {
 }
 
 // NewNativeClient creates a new Client using the golang ssh library
-func NewNativeClient(user, host, clientVersion string, port int, auth *Auth) (Client, error) {
-	if clientVersion == "" {
-		clientVersion = "SSH-2.0-Go"
+func NewNativeClient(user, hostname string, clientVersion []byte, auth *Auth) (Client, error) {
+	if clientVersion == nil {
+		clientVersion = []byte("SSH-2.0-Go")
 	}
 
-	config, err := NewNativeConfig(user, clientVersion, auth)
+	config, err := NewNativeConfig(user, hostname, string(clientVersion), auth)
 	if err != nil {
 		return nil, fmt.Errorf("Error getting config for native Go SSH: %s", err)
 	}
 
 	return &NativeClient{
 		Config:        config,
-		Hostname:      host,
-		Port:          port,
-		ClientVersion: clientVersion,
+		Hostname:      hostname,
+		ClientVersion: string(clientVersion),
 	}, nil
 }
 
 // NewNativeConfig returns a golang ssh client config struct for use by the NativeClient
-func NewNativeConfig(user, clientVersion string, auth *Auth) (ssh.ClientConfig, error) {
+func NewNativeConfig(user, host, clientVersion string, auth *Auth) (ssh.ClientConfig, error) {
```

Comment:
See other comment about not modifying exported function signatures for compatibility. (If you've a better suggestion/reasoning, we're open to hear it)